internal/cgroups/cgroups.go

```diff
@@ -104,7 +104,9 @@ func NewCGroups(procPathMountInfo, procPathCGroup string) (CGroups, error) {
 			if err != nil {
 				return err
 			}
-			cgroups[opt] = NewCGroup(cgroupPath)
+			if strings.HasPrefix(cgroupPath, "/sys") {
```

Comment:
Is there a way to test this without requiring the test to be executed on a container? The test is currently passing when running on my host, making it difficult to assert that the code is doing what's supposed to do.

Reply:
Sorry for taking so long to run a simple command, but here it is now:

```
> go test -v -run TestNewCGroups --count 5
=== RUN   TestNewCGroups
--- PASS: TestNewCGroups (0.00s)
=== RUN   TestNewCGroupsWithErrors
--- PASS: TestNewCGroupsWithErrors (0.00s)
=== RUN   TestNewCGroups
--- PASS: TestNewCGroups (0.00s)
=== RUN   TestNewCGroupsWithErrors
--- PASS: TestNewCGroupsWithErrors (0.00s)
=== RUN   TestNewCGroups
--- PASS: TestNewCGroups (0.00s)
=== RUN   TestNewCGroupsWithErrors
--- PASS: TestNewCGroupsWithErrors (0.00s)
=== RUN   TestNewCGroups
--- PASS: TestNewCGroups (0.00s)
=== RUN   TestNewCGroupsWithErrors
--- PASS: TestNewCGroupsWithErrors (0.00s)
=== RUN   TestNewCGroups
--- PASS: TestNewCGroups (0.00s)
=== RUN   TestNewCGroupsWithErrors
--- PASS: TestNewCGroupsWithErrors (0.00s)
PASS
ok  	go.opentelemetry.io/collector/internal/cgroups	0.002s
```

For the record, I'm running on Fedora 37, fully updated.

Reply:
Here it is:

```
=== RUN   TestNewCGroups
cgroupsSubsystems map[cpu:0xc00009d470 cpuacct:0xc00009d470 cpuset:0xc00009d4d0 memory:0xc00009d410]
mountPoint &{6 5 0:5 / /sys/fs/cgroup/cpuset [rw nosuid nodev noexec relatime] [shared:6] cgroup cgroup [rw cpuset]}
sybsys cpuset
cgroupPath /sys/fs/cgroup/cpuset
mountPoint &{7 5 0:6 /docker /sys/fs/cgroup/cpu,cpuacct [rw nosuid nodev noexec relatime] [shared:7] cgroup cgroup [rw cpu cpuacct]}
sybsys cpu
cgroupPath /sys/fs/cgroup/cpu,cpuacct
sybsys cpuacct
cgroupPath /sys/fs/cgroup/cpu,cpuacct
mountPoint &{8 5 0:7 /docker /sys/fs/cgroup/memory [rw nosuid nodev noexec relatime] [shared:8] cgroup cgroup [rw memory]}
sybsys memory
cgroupPath /sys/fs/cgroup/memory/large
mountPoint &{10 9 0:9 /docker /var/lib/docker/overlay2/9054a95f2cf7296867089e1bd37931742a17eb3308a795d51adb2654ee2276df/merged/sys/fs/cgroup/memory [ro nosuid nodev noexec relatime] [master:17] cgroup cgroup [rw memory]}
sybsys memory
cgroupPath /var/lib/docker/overlay2/9054a95f2cf7296867089e1bd37931742a17eb3308a795d51adb2654ee2276df/merged/sys/fs/cgroup/memory/large
    cgroups_test.go:70: 
        	Error Trace:	/home/jpkroehling/Projects/github.com/open-telemetry/opentelemetry-collector/internal/cgroups/cgroups_test.go:70
        	Error:      	Not equal: 
        	            	expected: "/sys/fs/cgroup/memory/large"
        	            	actual  : "/var/lib/docker/overlay2/9054a95f2cf7296867089e1bd37931742a17eb3308a795d51adb2654ee2276df/merged/sys/fs/cgroup/memory/large"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-/sys/fs/cgroup/memory/large
        	            	+/var/lib/docker/overlay2/9054a95f2cf7296867089e1bd37931742a17eb3308a795d51adb2654ee2276df/merged/sys/fs/cgroup/memory/large
        	Test:       	TestNewCGroups
        	Messages:   	"/sys/fs/cgroup/memory/large" expected for `cgroups["memory"].path`, got "/var/lib/docker/overlay2/9054a95f2cf7296867089e1bd37931742a17eb3308a795d51adb2654ee2276df/merged/sys/fs/cgroup/memory/large"
--- FAIL: TestNewCGroups (0.00s)
=== RUN   TestNewCGroupsWithErrors
cgroupsSubsystems map[]
cgroupsSubsystems map[]
cgroupsSubsystems map[cpu:0xc00009da40 cpuacct:0xc00009daa0]
mountPoint &{31 23 0:24 / /sys/fs/cgroup/cpu [rw nosuid nodev noexec relatime] [shared:1] cgroup cgroup [rw cpu]}
sybsys cpu
cgroupPath /sys/fs/cgroup/cpu/docker
mountPoint &{32 23 0:25 /docker/0123456789abcdef /sys/fs/cgroup/cpuacct [rw nosuid nodev noexec relatime] [shared:2] cgroup cgroup [rw cpuacct]}
sybsys cpuacct
--- PASS: TestNewCGroupsWithErrors (0.00s)
FAIL
exit status 1
FAIL	go.opentelemetry.io/collector/internal/cgroups	0.002s
```